src/cache/requests/dictionary/dictionary_get_fields.rs

```diff
@@ -0,0 +1,263 @@
+/// ```
+/// # use std::collections::HashMap;
+/// # use momento::cache::DictionaryGetFields;
+/// # use momento::MomentoResult;
+/// # let fetch_response = DictionaryGetFields::default();
+/// use std::convert::TryInto;
+/// let item: MomentoResult<HashMap<Vec<u8>, Vec<u8>>> = fetch_response.try_into();
+/// ```
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum DictionaryGetFields<F: IntoBytes> {
+    Hit {
+        fields: Vec<F>,
+        responses: Vec<DictionaryGetField>,
+    },
+    Miss,
+}
+
+impl Default for DictionaryGetFields<String> {
+    fn default() -> Self {
+        DictionaryGetFields::Hit {
+            fields: vec![],
+            responses: vec![],
+        }
+    }
+}
+
+impl<F: IntoBytes> TryFrom<DictionaryGetFields<F>> for HashMap<String, String> {
+    type Error = MomentoError;
+
+    fn try_from(value: DictionaryGetFields<F>) -> Result<Self, Self::Error> {
+        match value {
+            DictionaryGetFields::Hit {
+                fields, responses, ..
+            } => {
+                let mut result = HashMap::new();
+                for (field, response) in fields.into_iter().zip(responses.into_iter()) {
+                    match response {
+                        DictionaryGetField::Hit { value } => {
+                            let key: String = parse_string(field.into_bytes())?;
+                            let value: String = value.try_into()?;
+                            result.insert(key, value);
+                        }
+                        DictionaryGetField::Miss => (),
+                    }
+                }
+                Ok(result)
+            }
+            // In other SDKs we do not convert a `Miss` into an empty HashMap
+            DictionaryGetFields::Miss => Err(MomentoError {
+                message: "dictionary get fields response was a miss".into(),
+                error_code: MomentoErrorCode::Miss,
+                inner_error: None,
+                details: None,
+            }),
+        }
+    }
+}
+
+impl<F: IntoBytes> TryFrom<DictionaryGetFields<F>> for HashMap<Vec<u8>, Vec<u8>> {
+    type Error = MomentoError;
+
+    fn try_from(value: DictionaryGetFields<F>) -> Result<Self, Self::Error> {
+        match value {
+            DictionaryGetFields::Hit {
+                fields, responses, ..
+            } => {
+                let mut result = HashMap::new();
+                for (field, response) in fields.into_iter().zip(responses.into_iter()) {
+                    match response {
+                        DictionaryGetField::Hit { value } => {
+                            result.insert(field.into_bytes(), value.into());
+                        }
+                        DictionaryGetField::Miss => (),
+                    }
+                }
+                Ok(result)
+            }
+            // In other SDKs we do not convert a `Miss` into an empty HashMap
+            DictionaryGetFields::Miss => Err(MomentoError {
```

Comment:
when you rebase/merge you can use the `MomentoError::miss` here

Reply:
Done in [f6b3285](https://github.com/momentohq/client-sdk-rust/pull/247/commits/f6b3285c4d02bcb7fdd572313e6da1c96f2e7e00)